src/fuzz/fuzz.h

```diff
@@ -27,3 +27,6 @@ static inline bool outside_size_range(size_t size, size_t lower, size_t upper) {
                 return FUZZ_USE_SIZE_LIMIT;
         return false;
 }
+
+/* Force value to not be optimized away. */
+#define DO_NOT_OPTIMIZE(value) ({ asm volatile("" : : "g"(value) : "memory"); })
```

Comment:
hmm, couldn't we just make this explicit, i.e.:

```c
#pragma GCC push_options
#pragma GCC optimize ("O0")
…
#pragma GCC pop_options
```

if this works too, what are the pros and cons of that? as a pro i would see that we tell the compiler explicitly not to optimize, instead of relying on a side-effect?

Reply:
or instead, mark the whole function with `__attribute__((optimize("O0")))`

Reply:
clang does not understand these pragmas properly and they are ugly. And really you should only mark the stuff that you need/want as non-optimized, not the whole function. The asm method is so much nicer.